datalad/distribution/create.py

```diff
@@ -157,61 +162,69 @@ def __call__(
-                annex_version=annex_version,
-                annex_backend=annex_backend,
-                git_opts=git_opts,
-                annex_opts=annex_opts,
-                annex_init_opts=annex_init_opts)
 
-        else:
-            if no_annex:
-                if description:
-                    raise ValueError("Incompatible arguments: cannot specify "
-                                     "description for annex repo and declaring "
-                                     "no annex repo.")
-                if annex_opts:
-                    raise ValueError("Incompatible arguments: cannot specify "
-                                     "options for annex and declaring no "
-                                     "annex repo.")
-                if annex_init_opts:
-                    raise ValueError("Incompatible arguments: cannot specify "
-                                     "options for annex init and declaring no "
-                                     "annex repo.")
-
-                lgr.info("Creating a new git repo at %s", ds.path)
-                vcs = GitRepo(ds.path, url=None, create=True,
-                              git_opts=git_opts)
+            if sds is not None:
+                return sds.create_subdataset(
+                    ds.path,
+                    name=name,
+                    description=description,
+                    no_annex=no_annex,
+                    annex_version=annex_version,
+                    annex_backend=annex_backend,
+                    git_opts=git_opts,
+                    annex_opts=annex_opts,
+                    annex_init_opts=annex_init_opts)
             else:
-                # always come with annex when created from scratch
-                lgr.info("Creating a new annex repo at %s", ds.path)
-                vcs = AnnexRepo(ds.path, url=None, create=True,
-                                backend=annex_backend,
-                                version=annex_version,
-                                description=description,
-                                git_opts=git_opts,
-                                annex_opts=annex_opts,
-                                annex_init_opts=annex_init_opts)
-                # record the annex uuid of this repo for this afterlife
-                # to be able to track siblings and children
-                if 'datalad.annex.origin' in ds.config:
-                    # make sure we reset this variable completely, in case of a re-create
-                    ds.config.unset('datalad.annex.origin', where='dataset')
-                ds.config.add('datalad.annex.origin', vcs.uuid, where='dataset')
-                ds.repo.add('.datalad', git=True)
-
-            vcs.commit(msg="datalad initial commit",
-                       options=to_options(allow_empty=True))
-            # reset ID, we have a VCS now
-            ds._id = None
-            return ds
+                if isinstance(add_to_super, bool):
```

Comment:
hold on -- this logic is under `if add_to_super`, so wouldn't get here if `add_to_super=False`, so should be all kosher.  Otherwise it would have failed tests I bet - too common of a case ;)